homeassistant/components/samsungtv/bridge.py

```diff
@@ -115,23 +112,9 @@ def device_info(self) -> dict[str, Any] | None:
     def mac_from_device(self) -> str | None:
         """Try to fetch the mac address of the TV."""
 
-    def is_on(self) -> bool:
+    @abstractmethod
+    def is_on(self) -> bool | None:
         """Tells if the TV is on."""
-        if self._remote is not None:
-            self.close_remote()
-
-        try:
-            return self._get_remote() is not None
-        except (
-            UnhandledResponse,
-            AccessDenied,
-            ConnectionFailure,
-        ):
-            # We got a response so it's working.
-            return True
-        except OSError:
-            # Different reasons, e.g. hostname not resolveable
-            return False
```

Comment:
This method should always return immediately. Any work like this must be done in `update` or `async_update`.